### PR TITLE
fix test coverage.

### PR DESCRIPTION
**Pull request by nikhilmandlik**
Screen Shot 2020-04-17 at 1 51 31 AM

Screen Shot 2020-04-17 at 1 50 27 AM

### CODE REVIEW

File: karma.conf.js [hunk 1]
```diff
@@ -70,18 +77,37 @@ module.exports = (config) => {
...
         },
+        coverageIstanbulReporter: {
+            fixWebpackSourcePaths: true,
+            dir: process.env.CIRCLE_ARTIFACTS ?
+                process.env.CIRCLE_ARTIFACTS + '/coverage' :
+                "dist/reports/coverage",
+            reports: ['html', 'lcovonly', 'text-summary'],
+            thresholds: {
+                global: {
+                    statements: 50,
```
Comment (akhenry) [inline karma.conf.js:101]: Let's keep the `lines` threshold, and remove the others for now. Let's set it to 62.3 (the current level of coverage).

File: package.json [hunk 2]
```diff
@@ -84,6 +86,7 @@
     "build:watch": "webpack --watch",
     "test": "karma start --single-run",
     "test-debug": "cross-env NODE_ENV=debug karma start --no-single-run",
+    "test-coverage": "cross-env COVERAGE=true karma start --single-run",
```
Comment (akhenry) [inline package.json:89]: Let's simplify this and just always run coverage, whenever we run tests?

File: karma.conf.js [hunk 3]
```diff
@@ -24,16 +24,27 @@
 
 const devMode = process.env.NODE_ENV !== 'production';
 const browsers = [process.env.NODE_ENV === 'debug' ? 'ChromeDebugging' : 'ChromeHeadless'];
+const coverageEnabled = process.env.COVERAGE === 'true';
```
Comment (akhenry) [inline karma.conf.js:27]: I think we should just always run coverage when tests are run. No?

File: karma.conf.js (hunk 1 above)
Comment (nikhilmandlik) [reply karma.conf.js:101]: sure, np

**Comment by akhenry** [comment]
> Then I open `/dist/reports/coverage/HeadlessChrome 0.0.0 (Mac OS X 10.14.6)/index.html` and see the nice html coverage chart. When I was writing tests for NIRVSS I could then drill down into individual files by clicking on them and see exactly where in the code the tests were falling short. When I do so here, however, I see something like the following:
> Screen Shot 2020-04-17 at 8 19 30 AM

You *might* be looking at an old report. Mine produces a report in `/dist/reports/coverage/index.html`. When I drill down into the files reported here, they show the original source code and not the webpack'd source.

**Comment by JoelMcKinnon** [comment]
I'm seeing correct reports now. Not sure why it was messed up.

File: package.json (hunk 2 above)
Comment (nikhilmandlik) [reply package.json:89]: I added this for couple of reasons,
1. circle-ci: but now we want it to fail, all good here (need to increase size config.yml `NODE_OPTIONS=--max_old_space_size=4096`)
1. `npm test` is lot quicker than `npm run test-coverage`, so its nice to have option for quick pass/fail check

File: karma.conf.js (hunk 3 above)
Comment (nikhilmandlik) [reply karma.conf.js:27]: `npm test` is lot quicker than `npm run test-coverage`, so its nice to have option for quick pass/fail check

File: karma.conf.js (hunk 1 above)
Comment (nikhilmandlik) [reply karma.conf.js]: updated.

**Comment by nikhilmandlik** [comment]
> I'm seeing correct reports now. Not sure why it was messed up.

There was issue @JoelMcKinnon, I did fix it, but I guess your local was on old commit.

File: package.json (hunk 2 above)
Comment (nikhilmandlik) [reply package.json]: Or may be I can add another `npm run test-wo-coverage` and let `npm test` be with coverage?

File: karma.conf.js (hunk 3 above)
Comment (akhenry) [reply karma.conf.js:27]: @nikhilmandlik Fair enough!

File: package.json (hunk 2 above)
Comment (akhenry) [reply package.json]: @nikhilmandlik It's fine to keep them how they are. Good point re: performance. Could we rename `test-coverage` to `test:coverage`? To keep it consistent with other build targets. Let's also renamed `test:debug` while we're at it.

Comment (nikhilmandlik) [reply package.json]: @akhenry Do we need, circle ci to use `npm run test:coverage` instead of `npm run test`?

Comment (akhenry) [reply package.json]: > Do we need, circle ci to use npm run test:coverage instead of npm run test?

Yes, please.

File: karma.conf.js [hunk 4]
```diff
@@ -61,27 +68,27 @@ module.exports = (config) => {
...
             dir: process.env.CIRCLE_ARTIFACTS ?
                 process.env.CIRCLE_ARTIFACTS + '/coverage' :
                 "dist/reports/coverage",
-            check: {
+            reports: ['html', 'lcovonly', 'text-summary'],
+            thresholds: {
                 global: {
-                    lines: 80,
-                    excludes: ['src/plugins/plot/**/*.js']
+                    lines: 80
```
Comment (akhenry) [inline karma.conf.js:85]: I don't think we're this high. Set it to whatever our current coverage level is (61 or something from memory?). Basically I want circle-ci to fail if we fall below our current level.

Comment (nikhilmandlik) [reply karma.conf.js]: changed to `lines: 62`

**Comment by nikhilmandlik** [comment]
Testing Instructions:

1. run `npm run test:coverage` in terminal
1. Check test coverage summary in terminal
1. Check detail coverage for each file using `index.html` from `openmct/dist/reports/coverage`

**Pull request merged**